client/src/store/asyncAction.ts

```diff
@@ -7,17 +8,49 @@ import axios, { AxiosResponse } from "axios";
 export const fetchPosts = (): ThunkAction<void, RootState, unknown, AnyAction> => {
 	return async (dispatch): Promise<void> => {
 		dispatch(setIsFetchingPosts(true));
-		const response: AxiosResponse = await axios.get('http://localhost:4000/posts');
-		dispatch(getPosts(response.data));
+		try {
+			const response: AxiosResponse = await axios.get('http://localhost:4000/posts');
+			if (response.status === 200) {
```

Comment:
btw you can check for `response.ok`, it's boolean, 
in that case, 201 status, which is usually returned for successfully created entity, also will pass